src/changelog.rs

```diff
@@ -0,0 +1,146 @@
+//! Support for interpretting changelogs
+
+use camino::Utf8Path;
+
+use crate::errors::Result;
+use crate::{PackageInfo, Version, WorkspaceInfo};
+
+/// Info about a changelog entry
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone)]
+pub struct ChangelogInfo {
+    /// Title of the entry
+    pub title: String,
+    /// Body of the entry
+    pub body: String,
+}
+
+impl WorkspaceInfo {
+    /// Get the changelog for a given version from the root workspace's changelog
+    pub fn changelog_for_version(&self, version: &Version) -> Result<Option<ChangelogInfo>> {
+        if let Some(changelog_path) = self.root_auto_includes.changelog.as_deref() {
+            changelog_for_version(changelog_path, version)
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+impl PackageInfo {
+    /// Get the changelog for a given version from a package's changelog
+    pub fn changelog_for_version(&self, version: &Version) -> Result<Option<ChangelogInfo>> {
+        if let Some(changelog_path) = self.changelog_file.as_deref() {
+            changelog_for_version(changelog_path, version)
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+/// Get the changelog for a version
+pub fn changelog_for_version(
+    changelog_path: &Utf8Path,
+    version: &Version,
+) -> Result<Option<ChangelogInfo>> {
+    // Load and parse the changelog
+    let changelog_str = axoasset::LocalAsset::load_string(changelog_path)?;
+    changelog_for_version_inner(changelog_path, &changelog_str, version)
+}
+
+/// Get the changelog for a version (inner version for testing)
+pub fn changelog_for_version_inner(
+    changelog_path: &Utf8Path,
+    changelog_str: &str,
+    version: &Version,
+) -> Result<Option<ChangelogInfo>> {
+    let changelogs = parse_changelog::parse(changelog_str)?;
+
+    // Try to extract the changelog entry for the announcing version.
```

Comment:
this should be reworded since we're not dealing with "announcing" versions in a non-cargo-dist setting